src/platform/qt/library/LibraryController.h

```diff
@@ -47,16 +55,16 @@ struct LibraryEntry {
 	size_t filesize;
 	uint32_t crc32;
 
+	LibraryEntry& operator=(const LibraryEntry&) = default;
+	LibraryEntry& operator=(LibraryEntry&&) = default;
 	bool operator==(const LibraryEntry& other) const { return other.fullpath == fullpath; }
+
+	static inline uint64_t checkHash(size_t filesize, uint32_t crc32) { return (uint64_t(filesize) << 32) | crc32; }
```

Comment:
Though this seems potentially a bit of a weird way to add additional integrity to the hash, I suppose it's not the worst. Especially given how bad CRC32 is. (I intend to add more checksums down the line.) However, you should probably at least saturate the file size to 32 bits, just on the offhand chance someone somehow adds a 4 GB file (even though no ROMs are that big). A combination of clz and popcount is probably more than sufficient, though admittedly slower. There are utility functions for both in mgba-util/math.h--there might not be a 64-bit version of them atm but I believe there is on the medusa branch that can be backported. A basic single-round LCG is probably also sufficient.